### DISCUSSION TITLE
video stream in pocketbase

### DISCUSSION THREAD

**Discussion by UtkarshSiddhpura** [pocketbase/pocketbase#3439] [Q&A] [answered]
is it possible to create a video chat using pocketbase where one user can share his/her video stream to another user and vice versa?
(without webRTC)

**Comment by ganigeorgiev** [comment, marked as answer]
There is no builtin support for WebRTC or other video streaming APIs but you can implement it on your own using Go (_assuming that you want to use the Go server to relay the messages between peers_). You have full access to the PocketBase router (_[echo v5 alpha](https://echo.labstack.com/)_) when listening to the [`OnBeforeServe` hook](https://pocketbase.io/docs/go-event-hooks/#onbeforeserve) (_for more details you can also check the [Routing section](https://pocketbase.io/docs/go-routing/)_).
How this could be done will depend on your use case and what protocol you decide to use, eg. HLS, DASH, etc. (_for such cases usually ffmpeg is utilized but maybe there are better options_).

**Comment by rodydavis** [comment]
Created a signal server example with PocketBase and WebRTC!

https://github.com/rodydavis/webrtc-pocketbase-demo
